docs/upgrade/4.x.x.adoc

```diff
@@ -0,0 +1,475 @@
+= React4XP - Upgrade - 4.x.x
+:toc: right
+
+== What
+
+=== Graal.JS
+
+We have dropped support for Nashorn in favour of Graal.JS.
+
+This means we had to upgrade the Enonic XP version requirement to 7.12
+
+Since the Application Configuration setting `react4xp.ssr.settings` was only relevant for Nashorn, it has been removed.
+
+And since React4xp now only supports Graal.JS: the Application Configuration setting `react4xp.ssr.engineName` is no longer relevant.
+
+=== TypeScript
+
+React4xp version 4 introduces support for TypeScript
+
+* https://www.npmjs.com/package/@enonic/react4xp/v/4.0.0-B1
+* https://repo.enonic.com/service/rest/repository/browse/public/com/enonic/lib/lib-react4xp/4.0.0-B1
+* https://www.npmjs.com/package/@enonic/react-components/v/3.1.2
+* https://www.npmjs.com/package/@enonic-types/lib-react4xp/v/4.0.0-B2
```

Comment:
Why do we need links to all these repos here?

Reply:
We don't, see latest commit